services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/TableSchema.java

```diff
@@ -115,4 +172,17 @@
      * @return The {@link EnhancedType} of the modelled item this TableSchema maps to.
      */
     EnhancedType<T> itemType();
+
+    /**
+     * Returns a complete list of attribute names that are mapped by this {@link TableSchema}
+     */
+    List<String> attributeNames();
+
+    /**
+     * A boolean value that represents whether this {@link TableSchema} is abstract.
+     *
+     * @return true if it is abstract, and therefore cannot be used directly to create records but can be referred to
+     * by other schemata, and false if it is concrete and may be used to map records directly.
+     */
+    boolean isAbstract();
```

Comment:
Can we add more javadocs to explain the definition of `abstract` here since abstract is an overloaded term?

Reply:
Okay. +1